### ISSUE TITLE
macOS: deprecation error re: codecvt

### ISSUE THREAD

**Issue by bradgrantham-lunarg** [LunarG/gfxreconstruct#1420]
```
framework/util/json_util.cpp:182:31: error: 'codecvt_utf8<wchar_t>' is deprecated [-Werror,-Wdeprecated-declarations]
	std::wstring_convert<std::codecvt_utf8<wchar_t>> utf8_conv;
```
`codecvt` is deprecated and this line needs to be replaced for later macOS builds.  Not sure the version cutoff but it works on my machine with XCode 13 and fails on XCode 14 with XCode 15 CLI tools

Workaround is `--compile-no-warning-as-error` to `cmake`

**Comment by cdamus** [comment]
In addition to `--compile-no-warning-as-error` I found that I needed to comment out the `option(BUILD_WERROR "Build with warnings as errors" ON)` instruction in the `CMakeLists.txt`, which overrides that command-line option. Neither work-around, it seems, is sufficient on its own.

**Comment by bradgrantham-lunarg** [comment]
@richard-lunarg This is the existing issue for codecvt

**Comment by fabian-lunarg** [comment]
according to this discussion: https://stackoverflow.com/questions/42946335/deprecated-header-codecvt-replacement

- the situation seems a bit tricky. the entire <codecvt> header was deprecated in c++17, but no (stl-)alternative exists.
- according to https://isocpp.org/files/papers/p0636r0.html: "Users should use dedicated text-processing libraries instead", which would boil down to dependencies to icu, iconv, etc. ?
- why isn't this surfacing on other compilers/platforms?

for the moment:
- something like: ```#if defined(__clang__)
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
#endif
#include <codecvt>``` only in that compilation-unit seems a less crude workaround but still far from elegant.

basically more questions that answers here :)

**Comment by superwills** [comment]
@fabian-lunarg Experimenting, I found you can push/pop the warning setting, eg

```
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
// warnings suppressed
std::string getUtf8( const std::wstring &wstr ) {
  std::wstring_convert< std::codecvt_utf8<wchar_t>, wchar_t > convert;
  return convert.to_bytes( wstr );
}
#pragma clang diagnostic pop

// This function will still show warnings:
std::wstring fromUtf8( const std::string &str ) {
  std::wstring_convert< std::codecvt_utf8<wchar_t>, wchar_t > convert;
  return convert.from_bytes( str );
}
```

**Comment by bradgrantham-lunarg** [comment]
@superwills - I think that's what Fabian committed in #1493, without the guards, isn't it?  Except without the guards the #pragma will yield a warning itself on MSVC or gcc, so then you have to deal with that.  Better to wrap in the #ifdef guards, I think.

**Comment by superwills** [comment]
Apparently [you can use `std::filesystem::path` to do this type of conversion also](https://stackoverflow.com/a/77953816/111307) (and that hasn't been deprecated yet 😉)

**Comment by fabian-lunarg** [comment]
@superwills the push/pop is a good point and we did just in the related PR. as @bradgrantham-lunarg mentioned we're additionally guarding it with `#if defined(__clang__)`.

**Comment by fabian-lunarg** [comment]
the idea using std::filesystem::path sounds sneaky but might not even be applicable: 

https://en.cppreference.com/w/cpp/filesystem/path/string mentions for `wchar_t` (our case here):

"Otherwise, if path::value_type is wchar_t, conversion, if any, is unspecified."
